features/step_definitions/provider_steps.rb

```diff
@@ -147,6 +147,7 @@ def setup_provider(login)
   step 'I log in as provider "foo.3scale.localhost"' if login
 
   @provider = Account.find_by_domain!('foo.3scale.localhost')
+  @service ||= @provider.default_service
```

Comment:
Can service be already set? Maybe just `=`?

Reply:
Good point. I don't think it should be set here actually. This `@service` is reserved for steps that refer to a service. This one is about the provider.